// Copyright 2020 SWIR authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package swir_common;

enum HttpMethod{
  GET = 0;
  POST = 1;
  PUT = 2;
  DELETE = 3;
}

message InvokeRequest {
  string correlation_id=1;
  string service_name=2;
  HttpMethod method = 3;
  string request_target = 4;
  map<string, string> headers = 5;
  bytes payload = 6;
}

enum InvokeStatus {
  Ok = 0;
  Error = 1;
}

message InvokeResult {
  InvokeStatus status=1;
  string msg = 2;
}


message InvokeResponse {
  string correlation_id=1;
  string service_name=2;
  InvokeResult result=3;
  map<string, string> headers = 5;
  int32 status_code =6;
  bytes payload = 7;  
}

